test/built-ins/TypedArrays/internals/DefineOwnProperty/detached-buffer.js

```diff
@@ -0,0 +1,134 @@
+    b. If numericIndex is not undefined, then
+      ...
+      xi. If Desc has a [[Value]] field, then
+        1. Let value be Desc.[[Value]].
+        2. Return ? IntegerIndexedElementSet(O, intIndex, value).
+  ...
+
+  9.4.5.9 IntegerIndexedElementSet ( O, index, value )
+
+  ...
+  4. Let buffer be the value of O's [[ViewedArrayBuffer]] internal slot.
+  5. If IsDetachedBuffer(buffer) is true, throw a TypeError exception.
+  ...
+includes: [testTypedArray.js, detachArrayBuffer.js]
+features: [Reflect]
+---*/
+
+var desc = {
+  value: 0,
+  configurable: false,
+  enumerable: true,
+  writable: true
+};
+
+var obj = {
+  valueOf: function() {
+    throw new Test262Error();
+  }
+}
+
+testWithTypedArrayConstructors(function(TA) {
+  var sample = new TA([42]);
+  $DETACHBUFFER(sample.buffer);
+
+  assert.throws(TypeError, function() {
+    Reflect.defineProperty(sample, "0", desc);
+  }, "Throws TypeError on valid numeric index if instance has a detached buffer");
+
+  assert.sameValue(
+    Reflect.defineProperty(sample, "-1", desc),
+    false,
+    "Return false before Detached Buffer check when value is a negative number"
+  );
+
+  assert.sameValue(
+    Reflect.defineProperty(sample, "1.1", desc),
+    false,
+    "Return false before Detached Buffer check when value is not an integer"
+  );
+
+  assert.sameValue(
+    Reflect.defineProperty(sample, "-0", desc),
+    false,
+    "Return false before Detached Buffer check when value is -0"
+  );
+
+  assert.sameValue(
+    Reflect.defineProperty(sample, "0", {
+      configurable: true,
+      enumerable: true,
+      writable: true,
+      value: obj
+    }),
+    false,
+    "Return false before Detached Buffer check when desc configurable is true"
+  );
+
+  assert.sameValue(
+    Reflect.defineProperty(sample, "0", {
+      configurable: false,
+      enumerable: false,
+      writable: true,
+      value: obj
+    }),
+    false,
+    "Return false before Detached Buffer check when desc enumerable is false"
+  );
+
+  assert.sameValue(
+    Reflect.defineProperty(sample, "0", {
+      writable: false,
+      configurable: false,
+      enumerable: true,
+      value: obj
+    }),
+    false,
+    "Return false before Detached Buffer check when desc writable is false"
+  );
+
+  assert.sameValue(
+    Reflect.defineProperty(sample, "1", desc),
+    false,
+    "Return false before Detached Buffer check when key == [[ArrayLength]]"
+  );
+
+  assert.sameValue(
+    Reflect.defineProperty(sample, "2", desc),
+    false,
+    "Return false before Detached Buffer check when key > [[ArrayLength]]"
+  );
+
+  assert.sameValue(
+    Reflect.defineProperty(sample, "0", {
+      get: function() {}
+    }),
+    false,
+    "Return false before Detached Buffer check with accessor descriptor"
+  );
+
+  assert.sameValue(
+    Reflect.defineProperty(sample, "0", {
+      configurable: false,
+      enumerable: true,
+      writable: false
```

Comment:
In order for this assertion to pass, "writable" should be set to `true`. Also, it's probably safer to use a different property name for each assertion--implementers who make a mistake here could see misleading errors and/or false positives if these tests try to re-set the same property descriptor (given that it is sometimes being set to `configurable: false`).